repository_service_tuf_worker/repository.py

```diff
@@ -628,6 +633,76 @@ def bootstrap(
 
         return asdict(result)
 
+    def update_settings(
+        self,
+        payload: Dict[str, Any],
+        update_state: Optional[
+            Task.update_state
+        ] = None,  # It is required (see: app.py)
+    ) -> ResultDetails:
+        """Update repository settings with the new settings."""
+        tuf_settings: Dict[str, Any] = payload.get("settings")
+        if tuf_settings is None:
+            return ResultDetails(
+                status="update settings failed",
+                details={
+                    "update_settings": False,
+                    "message": "No 'settings' in the payload",
+                    "updated_roles": [],
+                    "non_valid_roles": [],
+                },
+                last_update=datetime.now(),
+            )
+
+        if tuf_settings.get("expiration") is None:
+            return ResultDetails(
+                status="update settings failed",
+                details={
+                    "update_settings": False,
+                    "message": "No 'expiration' in the payload",
+                    "updated_roles": [],
+                    "non_valid_roles": [],
+                },
+                last_update=datetime.now(),
+            )
```

Comment:
For the above three returns, I don't think we env need the updated and invalid roles in the details.